desc/objectives/_geometry.py

```diff
@@ -1393,3 +1393,593 @@ def compute(self, params, constants=None):
+        """Compute max absolute principal curvature.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of equilibrium or surface degrees of freedom,
+            eg Equilibrium.params_dict
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self.constants
+
+        Returns
+        -------
+        k : ndarray
+            Max absolute principal curvature at each point (m^-1).
+
+        """
+        if constants is None:
+            constants = self.constants
+
+        eq = self.things[0]
+        curve = self.things[1]
+
+        if self._curve_fixed:
+            curve_coords = constants["curve_coords"]
+            R = curve_coords["R"]
+            phi_arr = curve_coords["phi"]
+            Z = curve_coords["Z"]
+        else:
+            curve_data = compute_fun(
+                curve,
+                self._curve_data_keys,
+                params=params_2,
+                transforms=constants["curve_transforms"],
+                profiles={},
+            )
+            R = curve_data["R"]
+            phi_arr = curve_data["phi"]
+            Z = curve_data["Z"]
+
+        R = R.reshape(curve.NFP_umbilic_factor, 4 * curve.N).T
+        Z = Z.reshape(curve.NFP_umbilic_factor, 4 * curve.N).T
+
+        # axis location in rpz coordinates
+        data_axis = compute_fun(
+            eq,
+            self._equil_data_keys_axis,
+            params=params_1,
+            profiles=constants["equil_profiles_axis"],
+            transforms=constants["equil_transforms_axis"],
+        )
+
+        R_axis = data_axis["R"]
+        Z_axis = data_axis["Z"]
+
+        R_axis = R_axis.reshape(curve.NFP_umbilic_factor, 4 * curve.N).T
+        Z_axis = Z_axis.reshape(curve.NFP_umbilic_factor, 4 * curve.N).T
+
+        # calculate theta for each point in each section
+        theta = jnp.arctan2(Z - Z_axis, R - R_axis)
+        theta = jnp.mod(theta, 2 * jnp.pi)
+
+        phi_arr = jnp.mod(phi_arr, 2 * jnp.pi)
+
+        umbilic_edge_grid = Grid(
+            jnp.array([jnp.ones((len(phi_arr),)), theta.ravel(), phi_arr]).T,
+            jitable=True,
+        )
+
+        equil_profiles = get_profiles(
+            self._equil_data_keys,
+            obj=eq,
+            grid=umbilic_edge_grid,
+            has_axis=umbilic_edge_grid.axis.size,
+            jitable=True,
+        )
+        equil_transforms = get_transforms(
+            self._equil_data_keys,
+            obj=eq,
+            grid=umbilic_edge_grid,
+            has_axis=umbilic_edge_grid.axis.size,
+            jitable=True,
+        )
+
+        # now compute the curvature
+        data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._equil_data_keys,
+            params=params_1,
+            profiles=equil_profiles,
+            transforms=equil_transforms,
+        )
+
+        return data["curvature_k2_rho"]
```

Comment:
should we also multiply by like |e_theta x e_zeta| to weight by the local surface area? I am not sure